sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/HyperLogLogPlusPlus.scala

```diff
@@ -306,7 +307,7 @@ case class HyperLogLogPlusPlus(
     val estimate = if (V > 0) {
       // Use linear counting for small cardinality estimates.
       val H = m * Math.log(m / V)
-      if (H <= THRESHOLDS(p - 4)) {
+      if ((p < 19 && H <= THRESHOLDS(p - 4)) || E <= 2.5 * m) {
```

Comment:
I know the 2.5m bit is from the Flajolet paper but was it just missing here before?